build.sh

```diff
@@ -225,36 +256,71 @@ CMAKE_MORE_OPTIONS="${CMAKE_MORE_OPTIONS} ${CMAKE_PREFIX_PATH}"
 # Let's go
 # ---------------------------------------------------------------------------
 
+mkdir -p "$BUILD_DIR"
+
 cat <<EOF
 Darktable build script
 
 Building directory:  $BUILD_DIR
 Installation prefix: $INSTALL_PREFIX
 Build type:          $BUILD_TYPE
-Make program:        $MAKE
-Make tasks:          $MAKE_TASKS
+Build generator:     $BUILD_GENERATOR
+Build tasks:         $MAKE_TASKS
 
 
 EOF
 
-if [ ! -d "$BUILD_DIR" ]; then
-	mkdir "$BUILD_DIR"
+if [ $ADDRESS_SANITIZER -ne 0 ] ; then
+	ASAN_FLAGS="CFLAGS=\"-fsanitize=address -fno-omit-frame-pointer\""
+	ASAN_FLAGS="$ASAN_FLAGS CXXFLAGS=\"-fsanitize=address -fno-omit-frame-pointer\""
+	ASAN_FLAGS="$ASAN_FLAGS LDFLAGS=\"-fsanitize=address\" "
 fi
 
+
+cmd_config="${ASAN_FLAGS}cmake -G \"$BUILD_GENERATOR\" -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} -DCMAKE_BUILD_TYPE=${BUILD_TYPE} ${CMAKE_MORE_OPTIONS} \"$DT_SRC_DIR\""
+cmd_build="cmake --build "$BUILD_DIR" -- -j$MAKE_TASKS"
+cmd_install="${SUDO}cmake --build \"$BUILD_DIR\" --target install -- -j$MAKE_TASKS"
+
+
+OLDPWD="$(pwd)"
+
+if [ $DO_CONFIG -eq 0 ] ; then
+	cat <<EOF
+The script would have configured, built, and installed with these commands:
+\$ $(printf "$cmd_config")
+\$ $(printf "$cmd_build")
+\$ $(printf "$cmd_install")
+EOF
+	exit 0
+fi
+
+# configure the build
 cd "$BUILD_DIR"
+eval "$cmd_config"
+cd "$OLDPWD"
 
-cmake \
-	-DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} \
-	-DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
-	${CMAKE_MORE_OPTIONS} \
-	"$DT_SRC_DIR" \
-&& $MAKE -j $MAKE_TASKS
 
-if [ $? = 0 ]; then
+if [ $DO_BUILD -eq 0 ] ; then
+	cat <<EOF
+Darktable configuration is finished, to actually build and install darktable
```

Comment:
Would this be more readable presented on two lines? Also, darktable is always written with lowecase D.

> The darktable configuration is finished.
> To build and install darktable you need to type: